server/sonar-server/src/test/java/org/sonar/server/webhook/ws/WebhookDeliveryActionTest.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.rules.ExpectedException;
+import org.sonar.api.utils.System2;
+import org.sonar.api.web.UserRole;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbTester;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.webhook.WebhookDeliveryDto;
+import org.sonar.server.component.ComponentFinder;
+import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.exceptions.UnauthorizedException;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.WsActionTester;
+import org.sonarqube.ws.MediaTypes;
+import org.sonarqube.ws.Webhooks;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+import static org.sonar.db.webhook.WebhookDbTesting.newWebhookDeliveryDto;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class WebhookDeliveryActionTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+
+  private DbClient dbClient = db.getDbClient();
+  private WsActionTester ws;
+  private ComponentDto project;
+
+  @Before
+  public void setUp() {
+    ComponentFinder componentFinder = new ComponentFinder(dbClient);
+    WebhookDeliveryAction underTest = new WebhookDeliveryAction(dbClient, userSession, componentFinder);
+    ws = new WsActionTester(underTest);
+    project = db.components().insertComponent(newProjectDto().setKey("my-project"));
+  }
+
+  @Test
+  public void throw_UnauthorizedException_if_anonymous() {
+    expectedException.expect(UnauthorizedException.class);
+
+    ws.newRequest().execute();
+  }
+
+  @Test
+  public void return_404_if_delivery_does_not_exist() throws Exception {
+    userSession.login();
+
+    expectedException.expect(NotFoundException.class);
+
+    ws.newRequest()
+      .setMediaType(MediaTypes.PROTOBUF)
+      .setParam("deliveryId", "does_not_exist")
+      .execute();
+  }
+
+  @Test
+  public void load_the_delivery_of_example() throws Exception {
+    WebhookDeliveryDto dto = newWebhookDeliveryDto()
+      .setUuid("d1")
+      .setComponentUuid(project.uuid())
+      .setCeTaskUuid("task-1")
+      .setName("Jenkins")
+      .setUrl("http://jenkins")
+      .setCreatedAt(1_500_000_000_000L)
+      .setSuccess(true)
+      .setDurationMs(10)
+      .setHttpStatus(200)
+      .setPayload("{\"status\"=\"SUCCESS\"}");
+    dbClient.webhookDeliveryDao().insert(db.getSession(), dto);
+    db.commit();
+    userSession.login().addProjectUuidPermissions(UserRole.ADMIN, project.uuid());
+
+    String json = ws.newRequest()
+      .setParam("deliveryId", dto.getUuid())
+      .execute()
+      .getInput();
+
+    assertJson(json).isSimilarTo(IOUtils.toString(getClass().getResource("example-delivery.json")));
```

Comment:
I recenty found out that `ws.getDef().responseExampleAsString()` exists and it is quite convenient